src/LiveComponent/src/ComponentWithFormTrait.php

```diff
@@ -258,6 +258,20 @@ private function extractFormValues(FormView $formView): array
             if (\array_key_exists('checked', $child->vars)) {
                 // special handling for check boxes
                 $values[$name] = $child->vars['checked'] ? $child->vars['value'] : null;
+                continue;
+            }
+
+            if (\array_key_exists('choices', $child->vars)
+                && $child->vars['required']
+                && !$child->vars['multiple']
+                && !$child->vars['expanded']
+                && null === ($values[$name] ?? null)
+            ) {
+                if (0 === \count($choices = $child->vars['choices'])) {
+                    throw new UnprocessableEntityHttpException(\sprintf('The required field "%s" is missing and has no default value', $name));
+                }
+
+                $values[$name] = reset($choices)->value;
             } else {
```

Comment:
```suggestion
                continue;
            }

           // Special handling: required <select> fields with no value, no
            // placeholder and no multiple attribute should have their value set
            // to the first choice, if it exists.
            if (
                \array_key_exists('choices', $child->vars)
                && $child->vars['required']
                && !$child->vars['disabled']
                && !$child->vars['value']
                && !$child->vars['placeholder']
                && !$child->vars['multiple']
                && !$child->vars['expanded']
                && $child->vars['choices']
            ) {
                if (null !== $firstKey = array_key_first($child->vars['choices'])) {
                    $values[$name] = $child->vars['choices'][$firstKey]->value ?? null;
                    
                    continue;
                }
            }
            
```

There were some pre-condition checks missing. This one seems more robust, i'll let you check.